collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -130,6 +131,25 @@ namespace collision_detection
     }
   };
 
+  /** \brief Representation of a detailed distance */
+  struct DetailedDistance
+  {
+    DetailedDistance(std::string nearest_object, fcl::DistanceResult dist_result)
+    {
+      this->distance = dist_result.min_distance;
+      this->nearest_object = nearest_object;
+      this->nearest_points.first = Eigen::Vector3d(dist_result.nearest_points[0].data.vs);
+      this->nearest_points.second = Eigen::Vector3d(dist_result.nearest_points[1].data.vs);
+    }
+    DetailedDistance(){}
+
+    EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+    double distance; /**< Distance between objects */
+    std::string nearest_object; /**< Nearest object link name */
+    std::pair<Eigen::Vector3d, Eigen::Vector3d> nearest_points; /**< Point on each object */
```

Comment:
I original was going to use the fcl::DistanceResult structure but was not sure how to extract the information I needed about the object. I have since figured out how, this structure will be eliminated once I submit the changes.